test/google_crawler/search/search_keyword_worker_test.exs

```diff
@@ -0,0 +1,49 @@
+defmodule GoogleCrawler.Search.SearchKeywordWorkerTest do
+  use ExUnit.Case
+  use GoogleCrawler.DataCase
+
+  alias GoogleCrawler.SearchKeywordWorker
+  alias GoogleCrawler.Search
+  alias GoogleCrawler.KeywordFactory
+
+  test "search/1 performs search task with the given keyword id" do
+    keyword = KeywordFactory.create(%{})
+
+    task = SearchKeywordWorker.search(keyword.id)
+    task_ref = task.ref
+
+    assert %{^task_ref => {keyword, 0}} = SearchKeywordWorker.get_state()
+    assert Search.get_keyword(keyword.id).status == :in_progress
+
+    :timer.sleep(1000)
```

Comment:
How about adding the sleep in the Google API Client Mock to simulate delays?

Reply:
I think adding the delay to the client, we will need to put here also because we need to wait for the task to complete. (api is called in the task)

I found there is another way that we can monitor if the task is actually finish -> https://stackoverflow.com/questions/28745193/testing-asynchronous-code-in-elixir/28746505#28746505

Which I can do something like
```elixir
task = SearchKeywordWorker.search(keyword.id)

ref = Process.monitor(task.pid)
assert_receive {:DOWN, ^ref, :process, _, :normal}, 500
```

But because even if the task is completed, I do some db update in the genserver handle_info and I encounters the error 

![Screen Shot 2563-04-03 at 10 45 29](https://user-images.githubusercontent.com/6965195/78322215-4cab6080-7598-11ea-8117-5f1579991760.png)

I'm not completely sure what it is but maybe they're trying to update(in the genserver) and get the record(in the assertion). So when I use that, I added that I had to put antoher sleep before the assertion to make it work

```elixir
ref = Process.monitor(task.pid)
assert_receive {:DOWN, ^ref, :process, _, :normal}, 500
:timer.sleep(100) 
```
🤔 💭